dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -23,12 +23,134 @@ def self.extract_parent_emails
+      WHERE user_type = 'teacher' AND email > ''
+    SQL
+
+    # An user can have many user_geos records. user_geos records starts with only NULL
+    # values until a cronjob runs, does IP-to-address lookup, and update them later.
+    teacher_geos_query = <<~SQL
+      SELECT
+        teachers.email,
+        user_geos.city, user_geos.state, user_geos.postal_code, user_geos.country,
+        MAX(user_geos.updated_at) as updated_at
+      FROM (#{teacher_query}) AS teachers
+      JOIN user_geos
+      ON teachers.id = user_geos.user_id
+      GROUP BY email, city, state, postal_code, country
+    SQL
+
+    extraction_query = get_extraction_query(
+      teacher_geos_query,
+      'email',
+      %w(city state postal_code country),
+      true,
+      'dashboard.user_geos'
+    )
+    ContactRollupsV2.execute_query_in_transaction(extraction_query)
+  end
+
+  def self.extract_pd_enrollments
+    enrollment_email_query = <<~SQL
+      SELECT email, MAX(updated_at) AS updated_at
+      FROM pd_enrollments
+      WHERE email > ''
+      GROUP BY email
+    SQL
+
+    extraction_query = get_extraction_query(
+      enrollment_email_query,
+      'email',
+      [],
+      true,
+      'dashboard.pd_enrollments'
+    )
+    ContactRollupsV2.execute_query_in_transaction(extraction_query)
+  end
+
+  def self.extract_census_submissions
+    submitter_query = <<~SQL
+      SELECT submitter_email_address, submitter_role, MAX(updated_at) AS updated_at
+      FROM census_submissions
+      WHERE submitter_email_address > ''
+      GROUP BY submitter_email_address, submitter_role
+    SQL
+
+    extraction_query = get_extraction_query(
+      submitter_query,
+      'submitter_email_address',
+      ['submitter_role'],
+      true,
+      'dashboard.census_submissions'
+    )
+    ContactRollupsV2.execute_query_in_transaction(extraction_query)
+  end
+
+  def self.extract_school_geos
+    teacher_query = <<~SQL
+      SELECT email, school_info_id, updated_at
+      FROM users
+      WHERE user_type = 'teacher' AND email > ''
+    SQL
+
+    school_geos_query = <<~SQL
+      SELECT email, city, state, zip, MAX(updated_at) AS updated_at
+      FROM (
+        SELECT
+          teachers.email,
+          schools.city, schools.state, schools.zip,
+          GREATEST(teachers.updated_at, school_infos.updated_at, schools.updated_at) AS updated_at
+        FROM (#{teacher_query}) AS teachers
+        JOIN school_infos ON school_infos.id = teachers.school_info_id
+        JOIN schools ON schools.id = school_infos.school_id
+      ) AS inner_query
+      GROUP BY email, city, state, zip
+    SQL
+
+    extraction_query = get_extraction_query(
+      school_geos_query,
+      'email',
+      %w(city state zip),
+      true,
+      'dashboard.schools'
+    )
+    ContactRollupsV2.execute_query_in_transaction(extraction_query)
+  end
+
+  def self.extract_pegasus_form_geos
+    # TODO: how to run this method in Rails end-to-end tests? It reads from pegasus tables.
```

Comment:
🤔 good question...no clue.

Reply:
A temporary fix is to... not run pegasus extractions in test environments.

Reply:
Not pretty, but unblock us for a while.